include/fi_indexer.h

```diff
@@ -48,30 +48,30 @@
 
 union idx_entry {
 	void *item;
-	int   next;
+	int64_t next;
 };
 
-#define IDX_INDEX_BITS 16
-#define IDX_ENTRY_BITS 10
-#define IDX_ENTRY_SIZE (1 << IDX_ENTRY_BITS)
-#define IDX_ARRAY_SIZE (1 << (IDX_INDEX_BITS - IDX_ENTRY_BITS))
-#define IDX_MAX_INDEX  ((1 << IDX_INDEX_BITS) - 1)
+#define IDX_INDEX_BITS 32
+#define IDX_ENTRY_BITS 26
```

Comment:
This would end up allocating a 64 million entry array to store a single value.  This likely needs more than a 2-level table.  Some sort of trie of dynamic depth may be needed.